fairseq/options.py

```diff
@@ -609,6 +609,10 @@ def add_generation_args(parser):
                        help='sample from top K likely next words instead of all words')
     group.add_argument('--sampling-topp', default=-1.0, type=float, metavar='PS',
                        help='sample from the smallest set whose cumulative probability mass exceeds p for next words')
+    group.add_argument('--constraints', const="ordered", nargs="?", choices=["ordered", "unordered"],
+                       help='enables lexically constrained decoding')
+    group.add_argument('--constraints-both', action="store_true",
+                       help='translate sentence both constrained and unconstrained')
```

Comment:
how and why would users use this in practice? is this a common case? maybe its better to omit this + relevant code, and whoever is interested can just decode twice ?

Reply:
More a debugging option. Removed.